azurerm/resource_arm_cosmos_db_account.go

```diff
@@ -53,6 +53,18 @@ func resourceArmCosmosDBAccount() *schema.Resource {
 				}, true),
 			},
 
+			"kind": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```

Comment:
As this is a ForceNew field, are you sure we don't need a state migration? Technically the old state will have empty `kind` (or no `kind` to be precise) and we're setting default here, so `"" => "global-document-db"` would cause recreation, unless I'm mistaken?

Reply:
Just verified - seems good since there's a default value and this field was missing previously:

```
$ envchain azurerm terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

azurerm_resource_group.test: Refreshing state... (ID: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/tharvey-dev)
azurerm_cosmosdb_account.test: Refreshing state... (ID: /subscriptions/00000000-0000-0000-0000-...ntDB/databaseAccounts/tharvey-cosmosdb)
No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, Terraform
doesn't need to do anything.
```